### ISSUE TITLE
Bar charts silently ignore Group By, Facet Column and Facet Row

### ISSUE THREAD

**Issue by AshishK714** [social-engineer-ai/PyAnalytica#9]
**Labels:** bug, severity:medium, visualize, ux

## Environment

| | |
|---|---|
| PyAnalytica | 0.7.0 (installed from `dist/pyanalytica-0.7.0-py3-none-any.whl` into a clean venv) |
| Python | 3.14 |
| Key deps | shiny 1.7.0, pandas 3.0.5, numpy 2.5.2, scipy 1.18.1, matplotlib 3.11.1, seaborn 0.13.2 |
| OS / browser | Windows 11, Chrome |
| Dataset | `bank_campaign.csv` — 41,188 rows x 16 cols (UCI Bank Marketing, columns renamed) |

Found during a systematic 153-case sweep of every panel against one dataset.
Numbers were checked against an independent pandas/scipy/sklearn computation, not just
"did it render".

## Summary

With Chart Type = `bar`, the Group By / Facet Column / Facet Row selects remain visible and
accept values, but have no effect on the chart. The other three chart types honour them.

## Steps to reproduce

1. Load `bank_campaign.csv`.
2. Visualize > Distribute: Variable = `education`, Chart Type = **bar**,
   Facet Column = `channel`. Plot.

## Expected

Two faceted panels (Landline / Mobile), or the facet controls hidden/disabled for bar charts.

## Actual

A single un-faceted bar chart. The bar heights themselves are correct (they match
`value_counts`, including `illiterate` = 18), but the facet setting is silently discarded.
Same for Group By and Facet Row.

## Cause

`ui/modules/visualize/mod_distribute.py`:

```python
elif ct == "bar":
    fig, snippet = bar_chart(df, col, orientation=input.orientation(), pct=input.pct())
```

and `visualize/distribute.py`:

```python
def bar_chart(df, col, orientation="vertical", sort=True, pct=False):
```

`bar_chart()` does not accept `group_by`, `facet_col` or `facet_row` at all. `histogram()`,
`boxplot()` and `violin()` all take and use them, so the inconsistency is invisible until a
user tries it.

## Why this matters

A student comparing a categorical variable across channels sees one chart and reasonably
concludes there is nothing to compare — the tool gave no hint the split was dropped.

## Suggested fix

Either implement grouping/faceting for `bar_chart` (a `seaborn.catplot`-style path), or hide
the three controls when Chart Type is `bar` so the UI cannot promise what it will not do.

## Evidence

`shots/C14_DEFECT_bar_facet_ignored.jpg` — Facet Column = `channel` in the sidebar, one
un-faceted chart in the panel.

---

*Found in a systematic 153-case sweep of every panel against a 41,188-row teaching dataset, with each number checked against an independent pandas/scipy/scikit-learn computation rather than against the app itself.*
